contracts/ArrayLib.sol

```diff
@@ -101,20 +112,30 @@ library UInt256ArrayLib {
   ) internal pure returns (uint256[] memory ret) {
     assert(from <= to);
     assert(to <= arr.length);
+    uint256 len;
+    unchecked {
+      len = to - from;
+    }
+    ret = new uint256[](len);
     assembly {
-      ret := add(arr, mul(32, from))
-      mstore(ret, sub(to, from))
+      for {
+        let i := 1
+      } lt(i, add(len, 1)) {
+        i := add(i, 1)
+      } {
+        mstore(add(ret, mul(32, i)), mload(add(arr, mul(32, add(from, i)))))
+      }
     }
   }
 
   /**
    * @notice Copy an array.
    * @dev Copy an array `from` to an array `to`.
-   * @param from The array to copy from.
    * @param to The array to copy to.
+   * @param from The array to copy from.
    * @return The copied array.
    */
-  function copy(uint256[] memory from, uint256[] memory to)
+  function copy(uint256[] memory to, uint256[] memory from)
```

Comment:
this isn't currently used anywhere, but I updated it to be slightly more useful. This now copies like `brandNewEmptyArray.copy(arrayToCloneValuesFrom)`